test/stores/VDocument.test.ts

```diff
@@ -1088,5 +1088,223 @@ describe('stores', () => {
+                testEditor({
+                    contentBefore: '<p><b>ab[cd</b></p><p><b>ef<br/>gh</b>ij<i>kl]</i>mn</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p><b>ab[]</b>mn</p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p><b>ab]cd</b></p><p><b>ef<br/>gh</b>ij<i>kl[</i>mn</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p><b>ab[]</b>mn</p>',
+                });
+                // deleteBackward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p><b>ab[cd</b></p><p><b>ef<br/>gh</b>ij<i>kl]</i>mn</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p><b>ab[]</b>mn</p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p><b>ab]cd</b></p><p><b>ef<br/>gh</b>ij<i>kl[</i>mn</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p><b>ab[]</b>mn</p>',
+                });
+            });
+            it('should delete all contents of a complex DOM with format nodes and multiple paragraphs', () => {
+                // deleteForward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p><b>[abcd</b></p><p><b>ef<br/>gh</b>ij<i>kl</i>mn]</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>[]<br></p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p><b>]abcd</b></p><p><b>ef<br/>gh</b>ij<i>kl</i>mn[</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>[]<br></p>',
+                });
+                // deleteBackward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p><b>[abcd</b></p><p><b>ef<br/>gh</b>ij<i>kl</i>mn]</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>[]<br></p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p><b>]abcd</b></p><p><b>ef<br/>gh</b>ij<i>kl</i>mn[</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>[]<br></p>',
+                });
+            });
+            it('should delete a selection accross a heading1 and a paragraph', () => {
+                // deleteForward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<h1>ab [cd</h1><p>ef]gh</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<h1>ab []gh</h1>',
+                });
+                // Backward selection
```

Comment:
We might want to have a mechanism that "duplicates" the tests with a backward range to avoid having to write every tests twice ? Maybe something like `testBothRangeDirections: true` but with a better name... ;-)

Reply:
Good idea! Pin for a separate PR?